src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -720,6 +721,18 @@ public void DeleteRevision(DocumentsOperationContext context, Slice key, string
                 }
 
                 revisionEtag = TableValueToEtag((int)RevisionsTable.Etag, ref tvr);
+
+                if (table.IsOwned(tvr.Id) == false) 
+                {
+                    // We request to delete revision with the wrong collection
+                    var revision = TableValueToRevision(context, ref tvr);
+                    var currentCollection = CollectionName.GetCollectionName(revision.Data);
```

Comment:
there is a method in DocumentsStorage that gets you the collection from cache. Probably we should use it?

Reply:
done